marimbabot_behavior/src/marimbabot_behavior/behavior_node.py

```diff
@@ -279,13 +290,14 @@ def callback_command(self, command_msg):
 
         # read notes on the whiteboard
         if command_msg.behavior == "read":
-            rospy.loginfo('reading notes')
+            rospy.logdebug('reading notes')
             # update the note_sequence variable with the latest note sequence from vision_node/recognized_notes to signal that notes have been read
             try:
                 # wait for the vision node to publish a recognized note sequence
                 self.note_sequence = rospy.wait_for_message('vision_node/recognized_notes', String, timeout=5).data 
                 rospy.loginfo(f"recognized notes: {self.note_sequence}")
-                self.response_pub.publish('Notess recognized.')
+                self.response_pub.publish('Notes recognized.')
+                self.check_for_repeat()
                 self.update_hit_sequence()
```

Comment:
in this function call below, we also check for repeats

Reply:
You are right, but here
first 
`self.check_for_repeat()` is called which operates on `self.note_sequence`.
Then, ` self.update_hit_sequence()` is called. In this function, `self.hit_sequence = read_notes(self.note_sequence)` sends the new `self.note_sequence` to `read_notes` in which at some part  `create_staff_from_notes` is called. In this function, there is also a check for a repeat.
Therefore, we check whether there is a repeat in `self.note_sequence` twice.

Could you adjust that, so we just do it in the behavior_node? :)

Reply:
I removed all repeat logic from the interpreter. This should do the trick :+1:

Reply:
Nice, but do you agree with what I said? 
Because that was my thought process

Reply:
Yeah I do agree. My changes in the behavior would cause the interpreter to skip the second repeat check every time. You are right that these lines should be removed. Good catch